tests/test_bdist_pex.py

```diff
@@ -81,3 +83,52 @@ def test_pex_args_shebang_with_spaces():
 
 def test_pex_args_shebang_without_spaces():
   assert_pex_args_shebang('#!/usr/bin/python')
+
+
+def test_unwriteable_contents():
+  my_app_setup_py = dedent("""
```

Comment:
I usually have to do this to get dedent working as I intend:
```python
dedent("""\
.....
.....
""")
```

It likes this is working without the `\` in the first line though.

Reply:
Yes, without the `\` I get an initial newline which is fine for debugging output.